src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -2881,8 +2882,12 @@ private uint getJitFlags(ref CORJIT_FLAGS flags, uint sizeInBytes)
 #endif
             }
 
-            if (this.MethodBeingCompiled.IsNativeCallable)
+            if (targetArchitecture == TargetArchitecture.X86
```

Comment:
This condition is backwards.

It would be best to block R2R for NativeCallable methods on Windows x86 - it will safe us from dealing with the ABI mismatches if/once we get to do the Windows x86 case propertly.

```
if (this.MethodBeingCompiled.IsNativeCallable)
{
    if (targetArchitecture == TargetArchitecture.X86 
        && targetOS == TargetOS.Windows)
    {
        throw new ....;
    }
    flags.Set(CorJitFlag.CORJIT_FLAG_REVERSE_PINVOKE);
}
```

Reply:
It would be with ifdef, like:
```
if (this.MethodBeingCompiled.IsNativeCallable)
{
#if READYTORUN 
    if (targetArchitecture == TargetArchitecture.X86 
        && targetOS == TargetOS.Windows)
    {
        throw new ....;
    }
#endif
    flags.Set(CorJitFlag.CORJIT_FLAG_REVERSE_PINVOKE);
}
```

Reply:
Wow. That is embarrassing.

Reply:
Wait, why is the `READYTORUN` option relevant here? I wish I understood all these options/configurations the Crossge/ReadyToRun/... scenarios. Every time I read through this code I feel less and less confident that I understand any of it.

Reply:
We try to keep this file shareable with the full AOT compiler in CoreRT. The differences are done via ifdefs currently.